tools/entity/entity.ts

```diff
@@ -1,32 +1,164 @@
+
+    if (error) {
+      program.error(JSON.stringify(error.detail));
+    } else {
+      console.log(chalk.dim(JSON.stringify(data, null, 2)));
+      console.log(chalk.bold.green('\n✨ Success ✨'));
+    }
+  });
+
+program
+  .command('update_geo')
+  .description('Parses a shapefile and updates the geometry for an entity')
+  .argument('<entityID>', 'Entity ID')
+  .option('-f, --filePath <file>', 'Path to a zipfile containing the shapefile(s). e.g. /path/to/geo.zip')
+  .option('-n, --name <file>', 'Name of the region set. default: The value for regionType')
+  .requiredOption<components['schemas']['TypeRegionValue']>(
+    '-t, --regionType <TypeRegionValue>',
+    `Region type. ${typeRegionValueValues.join()}`,
+    (val) => {
+      if (typeRegionValueValues.indexOf(val as any) === -1) {
+        throw new Error(`bad region type. options: ${typeRegionValueValues.join()}`);
+      }
+
+      return val as components['schemas']['TypeRegionValue'];
+    },
+  )
+  .action(
+    async (
+      entityID,
+      {
+        filePath,
+        regionType,
+        name,
+      }: { filePath: string; regionType: components['schemas']['TypeRegionValue']; name: string },
+    ) => {
+      if (!entityID) {
+        program.error('Entity ID is required');
+      }
+      if (!filePath) {
+        program.error('Shapefile is required');
+      }
+
+      const regionName = name || regionType.toLowerCase();
+      const zipData = await readFile(filePath);
+      const geojson = await shp(zipData);
+      const geojsonArray = Array.isArray(geojson) ? geojson : [geojson];
+
+      const region_sets: components['schemas']['RegionSetPayload'][] = [
+        {
+          name: regionName,
+          type: regionType,
+          primary: false,
+          coverage: false,
+          juris: true,
+          regions: geojsonArray.reduce(
+            (acc: components['schemas']['RegionPayload'][], featureCollection) => {
+              featureCollection.features.forEach((f) => {
+                const baseProps = {
+                  crs: 4326,
+                  name: regionName,
+                  internal_id: regionName,
+                };
+
+                switch (f.geometry.type) {
+                  case 'Polygon':
+                    // the api only accepts MultiPolygon so convert the polygon.
+                    acc.push({
+                      ...baseProps,
+                      geometry: {
+                        type: 'MultiPolygon',
+                        coordinates: [f.geometry.coordinates],
+                      },
+                    });
+                    break;
+
+                  case 'MultiPolygon':
+                    acc.push({
+                      ...baseProps,
+                      geometry: {
+                        type: 'MultiPolygon',
+                        coordinates: f.geometry.coordinates,
+                      },
+                    });
+                    break;
+                }
+              });
```

Comment:
Because there are only two options, how do you feel about using a ternary here instead of a switch
```suggestion
                acc.push({
                  ...baseProps,
                  geometry: {
                    type: 'MultiPolygon',
                    coordinates: f.geometry.type === 'Polygon' ? [f.geometry.coordinates] : f.geometry.coordinates,
                  },
                });
```
or is the thinking there may be different types in the future?

Reply:
Geometry type can be any valid GeoJSON which includes geometry that is not compatible with `MultiPolygon`. I meant to put a default case with an error.  

The error case is added here - b9c7f7f